db/document.go

```diff
@@ -316,6 +316,22 @@ func (doc *Document) BodyBytes() ([]byte, error) {
 	return doc._rawBody, nil
 }
 
+func (doc *Document) GetMetaMap(userXattrKey string) (map[string]interface{}, error) {
```

Comment:
Can you add a comment describing what this function does?  The name is a bit ambiguous.